### PR TITLE
Add Bitbucket Server Version option

### LINKED ISSUE #326: Auto webhook creation broken for Bitbucket 6.x and earlier
## Description
Jenkins version - 2.22.1
Plugin version - 2.8.0
Bitbucket server and version - 5.4.1

When I perform a 'Scan Organization Folder Now', I expect webhooks to be registered for my repos but this does not occur and instead I see 400 errors in the logs.

```
Could not register hooks for sandbox/wordpress-app
com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRequestException: HTTP request error. Status: 400: .
HttpResponseProxy{HTTP/1.1 400  [X-AREQUESTID: *16U2BIYx857x12710733x0, X-ASEN: SEN-6051868, X-AUSERID: 9658, X-AUSERNAME: username, Cache-Control: no-cache, no-transform, Vary: X-AUSERNAME,Accept-Encoding, Transfer-Encoding: chunked, X-Content-Type-Options: nosniff, Content-Type: application/json;charset=UTF-8, Date: Fri, 10 Jul 2020 18:17:06 GMT, Connection: close] org.apache.http.client.entity.DecompressingEntity@325b29b7}
	at com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerAPIClient.doRequest(BitbucketServerAPIClient.java:1049)
	at com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerAPIClient.postRequest(BitbucketServerAPIClient.java:1002)
	at com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerAPIClient.postRequest(BitbucketServerAPIClient.java:990)
	at com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerAPIClient.registerCommitWebHook(BitbucketServerAPIClient.java:600)
	at com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookAutoRegisterListener.registerHook(WebhookAutoRegisterListener.java:177)
	at com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookAutoRegisterListener.registerHooks(WebhookAutoRegisterListener.java:133)
	at com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookAutoRegisterListener.access$000(WebhookAutoRegisterListener.java:56)
	at com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookAutoRegisterListener$1.doRun(WebhookAutoRegisterListener.java:102)
	at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:91)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by bitwiseman**
This works around incompatibilities between older versions of bitbucket and bitbucket v7.x.
Added an option to let users specify their version.

Fixes #326

### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/BitbucketServerVersion.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket.server;
+
+import hudson.model.ModelObject;
+
+public enum BitbucketServerVersion implements ModelObject {
+    VERSION_7 {
```
Comment (jetersen) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/BitbucketServerVersion.java:29]: This needs to be validated it works with JCasC.

Currently we have another open issue on Webhook implementation enum.

https://github.com/jenkinsci/configuration-as-code-plugin/issues/1440

**Comment by jetersen** [comment]
I would like to see JCasC tests added.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/BitbucketServerEndpoint.java [hunk 2]
```diff
@@ -134,12 +141,32 @@ public void setCallCanMerge(boolean callCanMerge) {
     }
 
     public boolean isCallChanges() {
-        return callChanges;
+        return callChanges && BitbucketServerVersion.VERSION_7.equals(serverVersion);
```
Comment (jansohn) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/BitbucketServerEndpoint.java:144]: The call changes options was introduced due to changes in Bitbucket Server 7 but the API call IMHO also works for Bitbucket Server 6. We are still running Bitbucket Server 6 and currently have both options enabled (call merge / call changes) and do not see any errors in the logs.
With these settings we wouldn't need to change anything once the Bitbucket Server is upgraded to 7.x.

TLDR: I do not think you need to restrict this option to Bitbucket Server 7.

Comment (bitwiseman) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/BitbucketServerEndpoint.java:144]: @jansohn 
If the API is not in Bitbucket 6.x we do not want try to make this call. I would say we don't want to all the setting to be true when it does nothing.   But it could also be argued that we shouldn't force values in ways that might be surprising.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/BitbucketServerVersion.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/BitbucketServerVersion.java:29]: @jetersen 
Agreed, this needs testing in general.
